### PR TITLE
* Add option to configure securityContext capabilities

### PR DESCRIPTION
**Pull request by linefeedse**
# What this PR does / why we need it

To comply with CIS kubernetes benchmark, PodSecurityPolicies should require that at least NET_RAW, preferrably ALL 
capabilities be dropped. In order to run Jenkins under such a PodSecurityPolicy, it must be possible to configure the capabilities of the jenkins deployment.

Signed-off-by: Tor Ake Fransson <tor-ake.fransson@linefeed.se>

# Which issue this PR fixes

*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*

- fixes #

# Special notes for your reviewer

# Checklist

- [x] [DCO](https://github.com/jenkinsci/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] CHANGELOG.md was updated

### CODE REVIEW

File: charts/jenkins/templates/jenkins-master-deployment.yaml [hunk 1]
```diff
@@ -76,6 +76,10 @@ spec:
 {{- if not (eq (int .Values.master.runAsUser) 0) }}
         fsGroup: {{ .Values.master.fsGroup }}
 {{- end }}
+{{- if .Values.master.securityContextCapabilities }}
+        capabilities:
+{{ toYaml .Values.master.securityContextCapabilities | indent 10 }}
+{{- end }}
```
Comment (torstenwalter) [inline charts/jenkins/templates/jenkins-master-deployment.yaml:82]:
```suggestion
         {{- if .Values.master.securityContextCapabilities }}
        capabilities:
          {{ toYaml .Values.master.securityContextCapabilities | nindent 10 }}
        {{- end }}
```

File: charts/jenkins/values.yaml [hunk 2]
```diff
@@ -142,6 +142,10 @@ master:
   # When setting runAsUser to a different value than 0 also set fsGroup to the same value:
   runAsUser: 1000
   fsGroup: 1000
+  # If you have PodSecurityPolicies that require dropping of capabilities as suggested by CIS K8s benchmark, put them here
+  securityContextCapabilities: false
```
Comment (torstenwalter) [inline charts/jenkins/values.yaml:146]:
```suggestion
  securityContextCapabilities: {}
```

Using empty block `{}` might be better than using `false`.

File: charts/jenkins/values.yaml [hunk 3]
```diff
@@ -142,6 +142,10 @@ master:
...
+  #  drop:
+  #    - NET_RAW
```
Comment (torstenwalter) [inline charts/jenkins/values.yaml:148]: which capabilities would you drop based on CIS benchmark? We could add all of them here in the comment.

**Review by torstenwalter — CHANGES_REQUESTED** [review]

File: charts/jenkins/templates/jenkins-master-deployment.yaml (hunk 1 above)
Comment (linefeedse) [reply charts/jenkins/templates/jenkins-master-deployment.yaml:82]: I agree with your suggestion about the indentation, but the lines above it aren't indented so it'll look strange unless I reindented the whole file. And i don't want to be that random dude on the internet that steps in and reindents other peoples code :-D

File: charts/jenkins/values.yaml (hunk 3 above)
Comment (linefeedse) [reply charts/jenkins/values.yaml:148]: Actually, they suggest NET_RAW or ALL in chapter 5.2.7, and that's all they suggest. If we drop ALL, then we can't bind ports (need NET_BIND_SERVICE for that), so we're going with just NET_RAW in our setup. Alternative is to drop ALL and then add back the one's you need, but that requires extensive testing.

File: charts/jenkins/templates/jenkins-master-deployment.yaml (hunk 1 above)
Comment (torstenwalter) [reply charts/jenkins/templates/jenkins-master-deployment.yaml:82]: You could do it I will take care of indenting the rest sooner or later. Just want to have test cases first to verify that nothing breaks.

**Comment by torstenwalter** [comment]
@linefeedse would you take care of resolving the conflict with master?

File: charts/jenkins/templates/jenkins-master-deployment.yaml (hunk 1 above)
Comment (linefeedse) [reply charts/jenkins/templates/jenkins-master-deployment.yaml]: I've altered the local indentation to the best of my ability

**Review by torstenwalter — APPROVED** [review]

**Pull request merged**